olive/model/utils/hf_utils.py

```diff
@@ -37,26 +33,42 @@ def load_hf_model_from_task(task: str, name: str, **kwargs):
     else:
         raise ValueError("unsupported transformers version")
 
-    model_class = {"pt": targeted_task["pt"]}
-    class_tuple = ()
-    class_tuple = class_tuple + model_class.get("pt", (AutoModel,))
-
+    class_tuple = targeted_task["pt"] or (AutoModel,)
     model = None
-    for model_class in class_tuple:
+    for i, model_class in enumerate(class_tuple):
         try:
             model = model_class.from_pretrained(name, **kwargs)
             logger.debug("Loaded model %s with name_or_path %s", model_class, name)
-            return model
-        except (OSError, ValueError):
+            break
+        except (OSError, ValueError) as e:
+            if i == len(class_tuple) - 1:
+                # len(class_tuple) == 1 covers most common tasks like text-generation, text-classification, etc
+                # error could be device OOM, device_map: "auto" not supported, etc
+
+                # len(class_tuple) > 1: not common - image-segmentation, conversational, etc
+                # there is no easy way to get tracebacks for earlier failures, so just raise from last
+                raise
             # the ValueError need to be caught since there will be multiple model_class for single task.
             # if the model_class is not the one for the task, it will raise ValueError and
             # next model_class will be tried.
-            continue
+            logger.debug(
+                "Failed to load model %s with name_or_path %s.\n kwargs: %s.\n Exception raised: %s",
+                model_class,
+                name,
+                kwargs,
+                e,
+            )
+
+    if model is None:
```

Comment:
we can safely remove the model None check and raise of ValueError?

Reply:
updated